### DISCUSSION TITLE
SOLECTRUS automatisiert abfragen

### DISCUSSION THREAD

**Discussion by schaufma** [solectrus/solectrus#3135] [Ideas]
Hallo Zusammen,

gibt es eine Möglichkeit, das Solectrus als MQTT Broker (nicht als Collector) fungiert? Also eher der umgekehrte Weg, da Solectrus ja Daten abfragt ...

Warum frage ich ... Ich besitze seit langer Zeit folgendes Gerät: https://lametric.com/de-DE/time/tech-specs ... Für dieses Teil gibt es eine App (https://apps.lametric.com/apps/my_data_diy__with_no-code_possibilities_/8942?product=market&market=de-DE), die relativ einfach "Daten" (ohne Programmierung) abfragen kann. Hierzu ist entweder eine lauffähiger MQTT Broker oder aber auch ein JSON, XML File, HTTP Poll oder WebSocket notwendig.

Ich würde gerne die Details natürlich weiterhin über die Weboberfläche von Solectrus anschauen, aber auch einzelne Werte auf dem Lametric Display "schnell" visualisieren. Also z.B. die aktuelle Strom-Erzeugung ... Ich zeige auf dem Lametric Display auch schon weitere Daten an, z.B. meine Heizstäbe und am Liebsten wäre mir eine JSON oder XML Datei die ich abfragen kann ;-)

Vielleicht ist das auch für den ein oder anderen interessant? Oder gibt es bereits eine JSON oder XML "irgendwo" und ich habe das in der Doku überlesen?

Viele Grüße und vielen Dank für ein Info im Voraus!

Markus

**Comment by ledermann** [comment]
Einen MQTT-Broker möchte ich nicht implementiert, das wäre ein erheblicher Aufwand. Ich denke, das braucht es aber auch gar nicht, zumindest für deinen Anwendungsfall.

SOLECTRUS selbst hat zwar keine API (also keinen JSON/XML-Endpunkt), aber die InfluxDB hat so etwas. Du müsstest nur deren Port `8086` öffnen (ist standardmäßig deaktiviert, siehe `docker-compose.yml`) und dann kannst du z.B. Abfragen wie diese hier machen:

```bash
curl --request POST 'INFLUX_HOST/api/v2/query?org=INFLUX_ORG' \
     --header 'Content-Type: application/vnd.flux' \
     --header 'Authorization: Token INFLUX_TOKEN' \
     --data 'from(bucket: "INFLUX_BUCKET") |> range(start: -5m) |> filter(fn: (r) => r._measurement == "INFLUX_MEASUREMENT_PV" and r._field == "inverter_power") |> keep(columns: ["_value"]) |> last()'
```

(die Platzhalter `INFLUX_HOST`, `INFLUX_ORG`, `INFLUX_TOKEN`, `INFLUX_BUCKET` und `INFLUX_MEASUREMENT_PV` vorher mit deinen Werten füllen)

Diese Abfrage holt die aktuelle PV-Erzeugung und liefert bei mir folgenden Output:

```csv
,result,table,_value
,_result,0,6499
```

Also 6499 Watt jetzt im Moment. Ob man die Ausgabe verbessern und minimales JSON oder XML herausbekommen kann, habe ich nicht weiter ausprobiert. Aber eventuell kommt Lametric schon damit klar.

Ich denke, das könnte bei dir funktionieren. Du müsstest dich allerdings mit der Abfragesyntax von Flux beschäftigen. Weitere Infos zur InfluxDB-Api hier:
https://docs.influxdata.com/influxdb/v2/api/#operation/PostQuery

**Comment by schaufma** [comment]
Hallo und vielen Dank für die schnelle Rückmeldung!

Ich schaue mir das mal an und probiere mich aus ... Das mit dem JSON/XML Endpunkt ist für mich absoluten Laien halt ideal ... Aber mal schauen, vielleicht bekomme ich das auch so hin...

Danke Dir!
